aicssegmentation/workflow/workflow.py

```diff
@@ -162,3 +167,126 @@ def is_done(self) -> bool:
+            raise ValueError("workflow_definition")
+        self._workflow_definition = workflow_definition
+
+        self.input_path = Path(input_dir)
+        if not self.input_path.exists():
+            raise ValueError("The input directory does not exist")
+
+        self.output_path = Path(output_dir)
+        # Creating an the output directory at output_dir if it does not exist already
+        if not self.output_path.exists():
+            if self.output_path.parent.exists():
+                self.output_path.mkdir(parents=True, exist_ok=True)
+            else:
+                raise ValueError(f"Output directory does not exist, and cannot be created at {self.output_path.parent}")
+
+        self.files_count: int = 0
+        self.failed_files: Dict[Path, str] = dict()
+        self._channel_index = channel_index
+
+    def is_valid_image(self, image_path: Path) -> bool:
+        """
+        Check if file at a given image_path is a valid image type we support (
+
+        Params:
+            image_path (Path): image to check
+
+        Returns:
+            (bool): True if all images are .tiff
+        """
+        if (image_path.suffix.lower() in SUPPORTED_FILE_EXTENSIONS):
+            return True
+        else:
+            return False
+
+    def process_all(self):
+        """
+        Process all images in the input_dir with the workflow_definition used to set up the BatchWorkflow
+
+        Params:
+            none
+
+        Returns:
+            none
+        """
+        files = [f for f in self.input_path.glob("**/*") if f.is_file]
+        # Currently will save files in same format as they are in the input path
+        for f in files:
+            self.files_count += 1
+            if self.is_valid_image(f):
+                read_image = AICSImage(f)
+                # read and format image in the way we expect
+                image_from_path = self.format_image_to_3d(read_image)
+                try:
+                    # Run workflow on image
+                    workflow = Workflow(self._workflow_definition, image_from_path)
+                    result = workflow.execute_all()
+                    AICSImage(result).save(self.output_path.joinpath(f.name))
+                except Exception as e:
+                    # Handle failures during workflow execution/save
+                    self.failed_files[f] = f"Failed during processing with error {e}"
+            else:
+                self.failed_files[f] = f"Unsupported image type {f.suffix}"
+        self._write_log_file()
+
+    def _write_log_file(self) -> bool:
+        """
+        Write a log file to the output folder.
+
+        Params:
+            image_path (Path): image to check
+
+        Returns:
+            (bool): True if all images are .tiff
+        """
+        with open(self.output_path.joinpath("log.txt"), "w") as f:
+            if self.files_count == 0:
+                f.write("no files were processed")
```

Comment:
I think it'd be less ambiguous to log something like "There were no files to process in the input directory" -- "no files were processed" could mean processing failed for some reason